tutorials/discord-weather-forecast.mdx

```diff
@@ -79,40 +79,52 @@ cargo shuttle init --serenity
 After running it you, should see the following generated in `src/lib.rs`:
 
 ```rust src/lib.rs
-use log::{error, info}; use serenity::async_trait; use
-serenity::model::channel::Message; use serenity::model::gateway::Ready; use
-serenity::prelude::\*; use shuttle_service::error::CustomError; use
-shuttle_service::SecretStore; use sqlx::PgPool;
+use anyhow::Context as _;
+use serenity::model::application::interaction::{Interaction, InteractionResponseType};
+use serenity::model::gateway::Ready;
+use serenity::model::prelude::command::CommandOptionType;
+use serenity::prelude::*;
+use serenity::{async_trait, model::prelude::GuildId};
+use shuttle_secrets::SecretStore;
+use tracing::{error, info};
+use weather::get_forecast;
 
 struct Bot;
 
-#[async_trait] impl EventHandler for Bot { async fn message(&self, ctx: Context,
-msg: Message) { if msg.content == "!hello" { if let Err(e) =
-msg.channel_id.say(&ctx.http, "world!").await { error!("Error sending message:
-{:?}", e); } } }
-
-    async fn ready(&self, _: Context, ready: Ready) {
-        info!("{} is connected!", ready.user.name);
```

Comment:
We should keep this function, it should be below the `async fn message` function in the `impl EventHandler for Bot` code block.